google/cloud/aiplatform/utils/gcs_utils.py

```diff
@@ -0,0 +1,162 @@
+            destination_file_uri = (
+                destination_uri.rstrip("/") + "/" + source_file_relative_posix_path
+            )
+            _logger.debug(f'Uploading "{source_file_path}" to "{destination_file_uri}"')
+            destination_blob = storage.Blob.from_string(
+                destination_file_uri, client=storage_client
+            )
+            destination_blob.upload_from_filename(filename=source_file_path)
+    else:
+        source_file_path = source_path
+        destination_file_uri = destination_uri
+        _logger.debug(f'Uploading "{source_file_path}" to "{destination_file_uri}"')
+        destination_blob = storage.Blob.from_string(
+            destination_file_uri, client=storage_client
+        )
+        destination_blob.upload_from_filename(filename=source_file_path)
+
+
+def stage_local_data_in_gcs(
+    data_path: str,
+    staging_gcs_dir: Optional[str] = None,
+    project: Optional[str] = None,
+    location: Optional[str] = None,
+    credentials: Optional[auth_credentials.Credentials] = None,
+) -> str:
+    """Stages a local data in GCS.
+
+    The file copied to GCS is the name of the local file prepended with an
+    "aiplatform-{timestamp}-" string.
+
+    Args:
+        data_path: Required. Path of the local data to copy to GCS.
+        staging_gcs_dir:
+            Optional. Google Cloud Storage bucket to be used for data staging.
+        project: Optional. Google Cloud Project that contains the staging bucket.
+        location: Optional. Google Cloud location to use for the staging bucket.
+        credentials: The custom credentials to use when making API calls.
+            If not provided, default credentials will be used.
+
+    Returns:
+        Google Cloud Storage URI of the staged data.
+
+    Raises:
+        RuntimeError: When source_path does not exist.
+        GoogleCloudError: When the upload process fails.
+    """
+    data_path_obj = pathlib.Path(data_path)
+
+    if not data_path_obj.exists():
+        raise RuntimeError(f"Local data does not exist: data_path='{data_path}'")
+
+    staging_gcs_dir = staging_gcs_dir or initializer.global_config.staging_bucket
+    if not staging_gcs_dir:
+        project = project or initializer.global_config.project
+        location = location or initializer.global_config.location
+        # Creating the bucket if it does not exist.
+        # Currently we only do this when staging_gcs_dir is not specified.
+        # The buckets that we create are regional.
+        # This prevents errors when some service required regional bucket.
+        # E.g. "FailedPrecondition: 400 The Cloud Storage bucket of `gs://...` is in location `us`. It must be in the same regional location as the service location `us-central1`."
+        # We are making the bucket name region-specific since the bucket is regional.
+        staging_bucket_name = project + "-staging-" + location
+        client = storage.Client(project=project, credentials=credentials)
```

Comment:
`credentials` should also be retrieved from `initializer.global_config` similar to `project` and `location`.

Reply:
Done.